src/qweb/qweb.ts

```diff
@@ -122,10 +122,17 @@ const UTILS: Utils = {
       .join("");
   },
   getComponent(obj) {
-    while (obj && !obj.hasOwnProperty("__owl__")) {
+    return this.getScope(obj, "__owl__");
+  },
+  getScope(obj, str: string) {
```

Comment:
this is only used to write to the scope. if you look at its uses, there is always some duplication.  I think that this should probably be changed to a function `writeToScope(obj, str: string, value: value)`, so it is much clearer what it does, and can be slightly optimized.

Reply:
WritetoScope as suggested cannot work with ++var, because var gets *assigned* to the current scope before resolving in which scope it should be written in.